packages/devtools_app/lib/src/screens/memory/framework/memory_controller.dart

```diff
@@ -65,19 +68,46 @@ class MemoryFeatureControllers {
 ///
 /// The controller should be recreated for every new connection.
 class MemoryController extends DisposableController
-    with AutoDisposeControllerMixin {
+    with
+        AutoDisposeControllerMixin,
+        OfflineScreenControllerMixin<MemoryScreenOfflineData> {
   MemoryController({
     DiffPaneController? diffPaneController,
     ProfilePaneController? profilePaneController,
   }) {
-    controllers = MemoryFeatureControllers(
-      diffPaneController,
-      profilePaneController,
+    isOffline =
+        offlineController.shouldLoadOfflineData(ScreenMetaData.memory.id);
+    if (!isOffline) {
+      _controllers = MemoryFeatureControllers(
+        diffPaneController,
+        profilePaneController,
+      );
+    }
+  }
+
+  late final bool isOffline;
+
+  @override
+  FutureOr<void> processOfflineData(MemoryScreenOfflineData offlineData) {
+    offlineData = offlineData;
+  }
+
+  @override
+  OfflineScreenData screenDataForExport() {
+    return OfflineScreenData(
+      screenId: ScreenMetaData.memory.id,
+      data: {
+        '$MemoryScreenOfflineData': MemoryScreenOfflineData(),
```

Comment:
just return an empty map for now:
`data: {}`

This will eventually be something like:
```
data: OfflineMemoryData(profile: {}, snapshots: [...], ...).toJson(),
```

Reply:
If the entire MemoryScreenOfflineData is serializable, why I cannot pass it as is?
It seems to be convenient to have all the data as one object.

Reply:
The value passed to data will be encoded as json here: https://github.com/flutter/devtools/blob/master/packages/devtools_app/lib/src/shared/offline_mode.dart#L71.

So I don't think the `MemoryScreenOfflineData` object will serialize well for `jsonEncode`.

Reply:
Added hello world code for it. It works.

Reply:
This fails on DartPad
![Screenshot 2024-03-12 at 1 14 55 PM](https://github.com/flutter/devtools/assets/43759233/0904c55f-600a-4dd7-b2bd-80060cf82ffd)